packages/client/hmi-client/src/components/models/tera-stratified-value-matrix.vue

```diff
@@ -73,13 +78,45 @@ const chosenRow = ref('');
 const valueToEdit = ref('');
 const editableCellStates = ref<boolean[][]>([]);
 
+const matrixExpressionsList: any = ref([]);
+
+const parametersValueList = computed(() =>
+	props.modelConfiguration.configuration?.semantics.ode.parameters.reduce((acc, val) => {
+		acc[val.id] = val.value;
+		return acc;
+	}, {})
+);
+
+watch(
+	() => [matrix.value, props.shouldEval],
+	async () => {
+		const output: any = [];
+		await Promise.all(
+			matrix.value.map(async (row) =>
+				Promise.all(
+					row.map(async (cell) => {
+						if (cell.value?.id) {
+							const matrixVal = await getMatrixValue(cell.value.id, props.shouldEval);
+							if (!output[cell.row]) {
+								output[cell.row] = [];
+							}
+							output[cell.row][cell.col] = matrixVal;
+						}
+					})
+				)
+			)
+		);
+		matrixExpressionsList.value = output;
+	}
+);
```

Comment:
Slight preference if we don't have multiple layers of promise alls, is it possible to do something like?

```
matrix.value.map(row => {
  row.map(cell => {
    matrixExpressionList.value[row][col] = await getMatrixValue(...)  
  })
})
```

Reply:
fixed